zookeeper-server/src/main/java/org/apache/zookeeper/client/FourLetterWordMain.java

```diff
@@ -101,14 +123,16 @@ public static String send4LetterWord(
                 : new InetSocketAddress(InetAddress.getByName(null), port);
             if (secure) {
                 LOG.info("using secure socket");
-                try (X509Util x509Util = new ClientX509Util()) {
-                    SSLContext sslContext = x509Util.getDefaultSSLContext();
-                    SSLSocketFactory socketFactory = sslContext.getSocketFactory();
-                    SSLSocket sslSock = (SSLSocket) socketFactory.createSocket();
-                    sslSock.connect(hostaddress, timeout);
-                    sslSock.startHandshake();
-                    sock = sslSock;
+                if (sslContext == null) {
+                    try (X509Util x509Util = new ClientX509Util()) {
+                        sslContext = x509Util.getDefaultSSLContext();
+                    }
```

Comment:
What's the point of this try block without catch clause?
It's a bit strange to me.

Reply:
The "try-with-resources" construct ensures that each resource declared in the try is closed at the end of the statement, which can help prevent resource leaks. In this case, it's ensuring that the X509Util instance is properly closed after use. It doesn't necessarily need a catch clause because it's about managing resources rather than catching and handling exceptions.

Reply:
Yeah. It's for closing the resource `X509Util`.